src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -1495,7 +1481,21 @@ public DetailAstImpl visitSimpleMethodCall(JavaLanguageParser.SimpleMethodCallCo
 
     @Override
     public DetailAstImpl visitLambdaExp(JavaLanguageParser.LambdaExpContext ctx) {
-        return flattenedTree(ctx);
+        final DetailAstImpl lambda = create(ctx.LAMBDA());
+        lambda.addChild(visit(ctx.lambdaParameters()));
+
+        final DetailAstImpl rightHandLambdaChild;
+        if (ctx.block() != null) {
```

Comment:
Isn't calling `ctx.block()` twice an overhead?

Reply:
Yes, cleaned this up

Reply:
@dreis2211 please resolve this item if you are good.

Reply:
Can't really resolve it, because you force-pushed and made this outdated. Looks good though